### PR TITLE
spk init config file should handle multiple env vars

### LINKED ISSUE #947: spk init config file does not handle multiple env vars in the same value correctly
**Describe the bug:**
For the purposes of integration test, I want to be able to specify in the config file:
```
hld_repository: "https://dev.azure.com/${env:AZDO_ORG}/${env:AZDO_PROJECT}/_git/${env:hld_dir}"
```
When I run `spk init` with a config file that contains multiple values defined like above, it only replaces the first occurrence of an env variable. 

**To Reproduce:**

**Expected behavior:**
It should replace every occurrence with its corresponding env variable. 

**Screenshots:**

**Additional context:**

**Does this require updates to documentation?:** No

### PR DESCRIPTION
**Pull request by samiyaakhtar**
Closes https://github.com/microsoft/bedrock/issues/947

### CODE REVIEW

**Comment by samiyaakhtar** [comment]
Integration tests are failing since my forked repo can't access variables in the pipeline. Smoke tests for PR are a new thing? @andrebriggs

File: src/config.ts [hunk 1]
```diff
@@ -73,6 +61,31 @@ export const loadConfigurationFromLocalEnv = <T>(configObj: T): T => {
...
+    if (matches.length > 1) {
+      if (process.env[matches[1]]) {
+        value = value.replace(matches[0], process.env[matches[1]] as string);
+      } else {
+        logger.error(`Env variable needs to be defined for ${matches[1]}`);
+        throw Error(
+          `Environment variable needs to be defined for ${matches[1]} since it's referenced in the config file.`
+        );
+      }
+      matches = regexp.exec(original as string);
```
Comment (dennisseah) [inline src/config.ts]: [change request]
we may have a chance to get into an infinite loop. I thought this liner should be
```
matches = regexp.exec(value);
```

File: src/config.ts [hunk 2]
```diff
@@ -73,6 +61,31 @@ export const loadConfigurationFromLocalEnv = <T>(configObj: T): T => {
...
 
+/**
+ * Updates env variable(s) from process.env
+ * Supports multiple per value
+ * @param value Value from config file to be replaced with env vars
+ */
+export const updateVariableWithLocalEnv = (value: string): string => {
+  const regexp = /\${env:([a-zA-Z_$][a-zA-Z_$0-9]+)}/g;
+  const original = value;
+  let matches = regexp.exec(value as string);
```
Comment (dennisseah) [inline src/config.ts]: can be just 
```
let matches = regexp.exec(value);
```
because value is already a string

**Review by dennisseah — CHANGES_REQUESTED** [review]

File: src/config.ts (hunk 1 above)
Comment (samiyaakhtar) [reply src/config.ts]: `value` is being updated inside the loop. Based on docs if you want to iterate through all matches with regex, you need to call the method on the original string over and over. It's not intuitive at all, agreed. ([reference](https://stackoverflow.com/a/432503))

**Review by dennisseah — APPROVED** [review]
LGTM

**Comment by mtarng** [comment]
is there an associated unit test for this? @samiyaakhtar

**Comment by samiyaakhtar** [comment]
@mtarng thanks for the reminder, added a quick one

**Review by mtarng — APPROVED** [review]

**Pull request merged**